enhancements/installer/inputs-from-stdin.md

```diff
@@ -0,0 +1,182 @@
+---
+title: inputs-from-sdtin
+authors:
+  - "@oglok"
+  - "@copejon"
+reviewers:
+  - "@fzdarsky"
+approvers:
+  - "@crawford"
+  - "@staebler"
+  - "@dgoodwin"
+creation-date: 2020-12-24
+last-updated: 2021-01-13
+status: provisional
+---
+
+# Inputs from STDIN
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In many occasions, users need to customize the install-config target and/or the manifests generated by the OpenShift Installer in a reproducible manner. The most popular tool to allow this type of customization is “Kustomize”, and it is already well-known by the Kubernetes ecosystem. The result of any “Kustomization” is sent to standard output as a stream of manifests.
```

Comment:
The problem is `openshift-install`'s assumption on file structure and naming when reading from the manifest directory.

Popular templating tools (kustomize, ksonnet, helm, ...) either output a single stream of manifests to stdout (from where it could be redirected to a single file, of course) or to a manifest directory, but then it's up to the tool how files get named (typically some string concatenation of the resource's GVKN) and organized into subdirs (typically: flat).

Means one cannot currently solve the common use case of templating manifests without additional scripting that reorganizes a tool's output to meet `openshift-install`'s requirements.

Reply:
Then let's consider as an alternative whether openshift-install can work without some of those assumptions on file structure and naming. If openshift-install is going to support streaming from STDIN, then it would need to rid itself of those assumptions for that input path anyway. So this is not a compelling argument to me to allow for reading from STDIN.